weather.py

```diff
@@ -0,0 +1,414 @@
+        self.extract_files()  # call function to extract weatherman zip to path
+
+        # All arguments will be checked to ensure multiple arguments accepted
+        if self.__highest_arguments is not None:  # check if -e argument was set
+            for highest_argument in self.__highest_arguments:  # catering multiple inputs argument for multiple reports
+                self.highest_argument_handler(highest_argument)  # Calculating results one input at a time
+
+        if self.__average_arguments is not None:  # check if -a argument was set
+            for average_argument in self.__average_arguments:
+                self.average_argument_handler(average_argument)
+
+        if self.__chart_form_arguments is not None:  # check if -c argument was set
+            for chart_form_argument in self.__chart_form_arguments:
+                self.chart_argument_handler(chart_form_argument)
+
+    # Method to extract weather files to specified path
+    def extract_files(self):
+        # Extract the weatherman.zip file to path
+        with zipfile.ZipFile("weatherfiles.zip", "r") as zip_ref:
+            zip_ref.extractall(self.__path)
+
+    # Method that calculates the read data
+    def highest_argument_handler(self, current_argument):
+
+        is_valid = self.validate_year(current_argument)
+
+        if is_valid:
+            is_file = self.files_exists_in_year(current_argument)
+
+            if is_file:
+                result = self.calculate_highest(current_argument)
+
+                # Function that prints report according to the mode provided
+                self.generate_report_highest(result)
+            else:
+                print(f"Files for the year {current_argument} are not in the system.")
+
+    def validate_year(self, date):
+
+        if len(date) != 4:  # To check if year is entered correctly
+            print("Wrong year Format entered")
+            return False
+        return True
+
+    def files_exists_in_year(self, date):
+        for month in self.months:  # Running code for all 12 months and opening files that exist
+            file = self.get_file_path(date, month)
+
+            if path.exists(file):
+                return True
+        return False
+
+    def get_file_path(self, year, month):
+        if month.isdigit():
+            mon = int(month)
+            if 0 < mon < 12:
+                mon = self.months[mon - 1]
+                return self.files_path_prefix + year + "_" + mon + ".txt"
+        else:
+            return self.files_path_prefix + year + "_" + month + ".txt"
+
+    def calculate_highest(self, date):
+        highest_temp = 0
+        highest_humidity = 0
+        lowest_temp = sys.maxsize
+        highest_date, lowest_date, humid_date = "", "", ""
+        exists = False
+
+        for month in self.months:  # Running code for all 12 months and opening files that exist
+            year = date
+
+            file = self.files_path_prefix + year + "_" + month + ".txt"
```

Comment:
Here we can use `get_file_path`.
```suggestion
            file = self.get_file_path(year, month)
```

Reply:
done